sympy/polys/ring_series.py

```diff
@@ -419,6 +418,61 @@ def rs_integrate(self, x):
         p1[e] = self[expv]/(expv[n] + 1)
     return p1
 
+def fun(p, f, *args):
+    """
+    Function of a multivariate series computed by substitution
+
+      p: multivariate series
+      f: method name or function
+      args[:-2] arguments of f, apart from the first one
+      args[-2] = iv: names of the series variables
+      args[-1] = prec: list of the precisions of the series variables
+
+    The case with f method name is used to compute tan and nth_root
+    of a multivariate series:
+
+      fun(p, tan, iv, prec)
+      compute _x.tan(iv, prec), then substitute _x with p
+
+      fun(p, nth_root, n, iv, prec)
+      compute (_x + p[0]).nth_root(n, '_x', prec)), then substitute _x
+      with p - p[0]
+
+    Examples
+    ========
+
+    >>> from sympy.polys.domains import QQ
+    >>> from sympy.polys.rings import ring
+    >>> R, x, y = ring('x, y', QQ)
+    >>> p = x + x*y + x**2*y + x**3*y**2
+    >>> fun(p, _tan1, 'x', 4)
+    1/3*x**3*y**3 + 2*x**3*y**2 + x**3*y + 1/3*x**3 + x**2*y + x*y + x
```

Comment:
I'm sorry, I will update it.
`fun` is used only for calculating multivariate series expansion. It calculates the series for a dummy variable `_x` and replaces `_x` with the intended variable `p`.